curvefs/src/client/s3/disk_cache_write.cpp

```diff
@@ -283,6 +283,8 @@ int DiskCacheWrite::AsyncUploadFunc() {
         }
         toUpload.clear();
         if (GetUploadFile("", &toUpload) <= 0) {
+            std::unique_lock<std::mutex> lock(mtx_);
```

Comment:
`mutex_` is used to protect `waitUpload_`, but here we nothing about `waitUpload_`. `waitUpload_` may not satisfy the condition `waitUpload_.empty()`.

Reply:
well, check the `waitUpload_` after `std::unique_lock<std::mutex> lock(mtx_);`, if false, we will continue the loop;